### PR TITLE
LibGUI: Support applying AutocompleteBox suggestions with mouse

### PR DESCRIPTION
**Pull request by guerinoni**
Fixes: #8004

### CODE REVIEW

File: Userland/Libraries/LibGUI/AutocompleteProvider.cpp [hunk 1]
```diff
@@ -100,6 +100,14 @@ AutocompleteBox::AutocompleteBox(TextEditor& editor)
     m_suggestion_view = main_widget.add<GUI::TableView>();
     m_suggestion_view->set_column_headers_visible(false);
     m_suggestion_view->set_visible(false);
+    m_suggestion_view->on_activation = [&](const GUI::ModelIndex& index) {
```
Comment (linusg) [inline Userland/Libraries/LibGUI/AutocompleteProvider.cpp:103]: Please use east const

File: Userland/Libraries/LibGUI/AutocompleteProvider.cpp [hunk 2]
```diff
@@ -100,6 +100,14 @@ AutocompleteBox::AutocompleteBox(TextEditor& editor)
...
+        if (m_suggestion_view->model()->is_within_range(index)) {
+            m_suggestion_view->selection().set(index);
+            m_suggestion_view->scroll_into_view(index, Orientation::Vertical);
+        }
+
+        this->apply_suggestion();
```
Comment (linusg) [inline Userland/Libraries/LibGUI/AutocompleteProvider.cpp:109]: no need for `this->`

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibGUI/AutocompleteProvider.cpp (hunk 1 above)
Comment (guerinoni) [reply Userland/Libraries/LibGUI/AutocompleteProvider.cpp:103]: Sorry Linus :), I forgot to lint

File: Userland/Libraries/LibGUI/AutocompleteProvider.cpp (hunk 2 above)
Comment (guerinoni) [reply Userland/Libraries/LibGUI/AutocompleteProvider.cpp]: Done

**Comment by guerinoni** [comment]
@linusg friendly ping :)

File: Userland/Libraries/LibGUI/AutocompleteProvider.cpp [hunk 3]
```diff
@@ -100,6 +100,14 @@ AutocompleteBox::AutocompleteBox(TextEditor& editor)
     m_suggestion_view = main_widget.add<GUI::TableView>();
     m_suggestion_view->set_column_headers_visible(false);
     m_suggestion_view->set_visible(false);
+    m_suggestion_view->on_activation = [&](GUI::ModelIndex const& index) {
+        if (m_suggestion_view->model()->is_within_range(index)) {
+            m_suggestion_view->selection().set(index);
+            m_suggestion_view->scroll_into_view(index, Orientation::Vertical);
+        }
+
+        apply_suggestion();
```
Comment (linusg) [inline Userland/Libraries/LibGUI/AutocompleteProvider.cpp:109]: can this be done inside the if? If so, the if could be inverted (`!is_within_range()`) and do an early return for less nesting.

**Review by linusg — COMMENTED** [review]
We recently updated the guidelines for commit messages, inspired by this PR actually :)
Please update it to match https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md#code-submission-policy, something like

> LibGUI: Support applying AutocompleteBox suggestions with mouse

File: Userland/Libraries/LibGUI/AutocompleteProvider.cpp (hunk 3 above)
Comment (guerinoni) [reply Userland/Libraries/LibGUI/AutocompleteProvider.cpp:108]: Thank you for pointing me out the new guidelines :)
Now all should be fine

**Pull request merged**